File: Data Structures & Algorithms/search-insert-position/submission-1.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int n=nums.size(), low=0, high=n;
        while(low<high){
            int mid=low+(high-low)/2;
            if(nums[mid]<target){ //search for first num>=target
                low=mid+1; //search to right
            } else high=mid; //search to left
            
        } return low;
    }
};

File: Data Structures & Algorithms/majority-element/submission-1.java
class Solution {
    public int majorityElement(int[] nums) {
        int candi = nums[0];
        int count=0;

        for(int num:nums){
            if(count==0) candi=num;
            count+=(num==candi)?1:-1;
        }
        return candi;
    }
}